tools/websoc-scraper-v2/index.ts

```diff
@@ -414,44 +420,41 @@ async function scrape(name: string, term: Term) {
 /**
  * The entry point of the program.
  */
-(async function main() {
+async function main() {
   try {
-    logger.info("websoc-scraper-v2 daemon starting");
-    let now = new Date();
-    let termsInScope = await getTermsToScrape(now);
-    for (const [name, term] of Object.entries(termsInScope))
-      await scrape(name, term);
-    for (;;) {
-      const curr = new Date();
-      // Check the available terms every day and scrape all of them once.
-      if (curr.getDate() !== now.getDate()) {
-        termsInScope = await getTermsToScrape(curr);
-        for (const [name, term] of Object.entries(termsInScope))
-          await scrape(name, term);
-      }
-      now = curr;
-      // Check the database for terms to scrape on demand.
-      const websocTerms = await prisma.websocTerm.findMany({
-        where: { timestamp: { gte: now } },
-        select: { year: true, quarter: true },
-      });
-      const termsOnDemandEntries = websocTerms.map(
-        (x) => [`${x.year} ${x.quarter}`, x] as TermEntry
-      );
-      const termsOnDemand = Object.fromEntries(termsOnDemandEntries);
-      let scraped = false;
-      for (const [name, term] of Object.entries(termsOnDemand)) {
-        if (!(name in termsInScope)) {
-          logger.info(`Term ${name} requested but not in scope, skipping`);
-          continue;
-        }
-        scraped = true;
+    logger.info("websoc-scraper-v2 starting");
+    const prev = new Date(
+      existsSync("/tmp/websoc_scraper_v2")
+        ? parseInt(readFileSync("/tmp/websoc_scraper_v2").toString())
```

Comment:
```suggestion
        ? parseInt(readFileSync("/tmp/websoc_scraper_v2").toString(), 10)
```
lol.

An alternative that has some overhead, but might bit marginally nicer is `z.coerce.number().parse(...)`, which can parse numeric strings into numbers.